MHArmory.Search/OpenCL/HostOptionsBuilder.cs

```diff
@@ -0,0 +1,29 @@
+using System.Text;
+
+namespace MHArmory.Search.OpenCL
+{
+    class HostOptionsBuilder
+    {
+        private StringBuilder InnerBuilder { get; }
+
+        public HostOptionsBuilder()
+        {
+            InnerBuilder = new StringBuilder();
+        }
+
+        public void AddDefine(string name)
+        {
+            InnerBuilder.Append($"-D {name} ");
+        }
+
+        public void AddDefine(string name, object value)
+        {
+            InnerBuilder.Append($"-D {name}={value} ");
+        }
+
+        public override string ToString()
+        {
+            return InnerBuilder.ToString(0, InnerBuilder.Length - 1);
```

Comment:
Here `InnerBuilder.ToString()` is enough.

Reply:
Kind of, but not really. When I add options, I append the options to the internal StringBuilder with a trailing space, so that the next addition would be separated. Yes, the trailing space doesn't really harm anything, but it's just cleaner if it isn't there, so I skip the last character when returning the final string.

Reply:
OK, no problem then :)